src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php

```diff
@@ -73,6 +87,9 @@ public function getChildren()
             if ($children instanceof self) {
                 // parent method will call the constructor with default arguments, so unreadable dirs won't be ignored anymore
                 $children->ignoreUnreadableDirs = $this->ignoreUnreadableDirs;
+                $children->subPath = $children->getSubPath();
```

Comment:
Please add a command before this to explain it is a performance optimization to avoid redoing the same work in all childrens